python/nav/web/api/v1/auth.py

```diff
@@ -58,6 +62,56 @@ def authenticate(self, request):
             return request.account, None
 
 
+class JWTToken(dict):
+    pass
+
+
+class JWTAuthentication(JSONWebTokenAuthentication):
+    REQUIRED_CLAIMS = ["exp", "nbf", "aud", "iss"]
+    SUPPORTED_ALGORITHMS = ["RS256", "RS384", "RS512"]
+
+    def authenticate(self, request):
+        auth_result = super().authenticate(request)
+        if auth_result:
+            user, token = auth_result
+            return user, JWTToken(token)
+        else:
+            return None
+
+    def decode_jwt(self, jwt_value):
+        kid = self.get_kid(jwt_value)
+        try:
+            validated_token = jwt.decode(
+                jwt=jwt_value,
+                algorithms=self.SUPPORTED_ALGORITHMS,
+                key=self.get_public_key(kid),
+                options={"require": self.REQUIRED_CLAIMS},
+                audience=api_settings.OIDC_CLAIMS_OPTIONS['aud']['value'],
+                issuer=self.issuer,
+            )
+            return validated_token
+        except jwt.exceptions.PyJWTError as e:
+            raise AuthenticationFailed(f"Error validating token: {e}")
+
+    def get_kid(self, token):
```

Comment:
docstrings docstrings :)

I had to read the docs for `jwt.get_unverified_header` to understand what this was all about :)